aiida/backends/djsite/db/migrations/0031_remove_legacy_workflows.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name,too-few-public-methods
+"""Remove legacy workflow."""
+from __future__ import unicode_literals
+
+from __future__ import absolute_import
+import sys
+import click
+
+# Remove when https://github.com/PyCQA/pylint/issues/1931 is fixed
+# pylint: disable=no-name-in-module,import-error
+from django.core import serializers
+from django.db import migrations
+
+from aiida import settings
+from aiida.backends.djsite.db.migrations import upgrade_schema_version
+from aiida.common import json
+from aiida.cmdline.utils import echo
+
+REVISION = '1.0.31'
+DOWN_REVISION = '1.0.30'
+
+
+def export_workflow_data(apps, _):
+    """Export existing legacy workflow data to a JSON file."""
+    from tempfile import NamedTemporaryFile
+
+    DbWorkflow = apps.get_model('db', 'DbWorkflow')
+    DbWorkflowData = apps.get_model('db', 'DbWorkflowData')
+    DbWorkflowStep = apps.get_model('db', 'DbWorkflowStep')
+
+    count_workflow = DbWorkflow.objects.count()
+    count_workflow_data = DbWorkflowData.objects.count()
+    count_workflow_step = DbWorkflowStep.objects.count()
+
+    # Nothing to do if all tables are empty
+    if count_workflow == 0 and count_workflow_data == 0 and count_workflow_step == 0:
+        return
+
+    if not settings.TESTING_MODE:
+        echo.echo('\n')
+        echo.echo_warning('The legacy workflow tables contain data but will have to be dropped to continue.')
+        echo.echo_warning('If you continue, the content will be dumped to a JSON file, before dropping the tables.')
+        if not click.confirm('Are you sure you want to continue', default=True):
+            sys.exit(1)
+
+    delete_on_close = settings.TESTING_MODE
+
+    data = {
+        'workflow': serializers.serialize('json', DbWorkflow.objects.all()),
+        'workflow_data': serializers.serialize('json', DbWorkflowData.objects.all()),
+        'workflow_step': serializers.serialize('json', DbWorkflowStep.objects.all()),
+    }
+
+    with NamedTemporaryFile(
+            prefix='legacy-workflows', suffix='.json', dir='.', delete=delete_on_close, mode='w+') as handle:
+        filename = handle.name
+        json.dump(data, handle)
+
+    # If delete_on_close is False, we are running for the user and add additional message of file location
+    if not delete_on_close:
+        echo.echo_info('Exported workflow data to {}'.format(filename))
+
+
+class Migration(migrations.Migration):
+    """Remove legacy workflow."""
+
+    dependencies = [
+        ('db', '0030_dbnode_type_to_dbnode_node_type'),
+    ]
+
+    operations = [
+        # Export existing data to a JSON file
+        migrations.RunPython(export_workflow_data, reverse_code=migrations.RunPython.noop),
+        migrations.RemoveField(
+            model_name='dbworkflow',
+            name='user',
+        ),
+        migrations.AlterUniqueTogether(
```

Comment:
I guess this is automatically generated - interesting that it first removes each field

Reply:
It was indeed auto generated by `makemigrations`